### PR TITLE
✨ os: add sudo resource for sudo binary metadata

### PR DESCRIPTION
**Pull request by tas50**
## Summary

- Adds a new `sudo` resource (binary-side) that complements the existing `sudoers` resource (config-side)
- Exposes `version`, `path`, `installed`, `plugins[]` (with `policyPlugin` / `ioPlugins` / `auditPlugins` / `approvalPlugins`), `pythonSupport`, `validate()`, and a `sudoers` shorthand
- Parses `sudo -V` once per resource instance (`sync.Once`) and shares the result across all derived fields
- Auto-detects the sudo binary across Linux / AIX / Solaris common install paths with a `command -v` fallback
- `validate()` runs `visudo -c` and reports parse errors via `sudo.validation` / `sudo.validation.error`; returns null when the sudoers files are unreadable (unprivileged session) so policies can distinguish "validation unavailable" from "validation failed"
- Designed to replace `command("sudo --version")` shellouts in CIS AIX 7.x policies

## Test plan

- [x] Parser unit tests: 22 cases — version forms (1.8.x, 1.9.5p3, 1.9.5p11, leading whitespace), all four plugin families (policy / io / audit / approval), Python plugins, Python detection via configure options, I/O → io normalization, distro-patched plugin versions, blank/comment-line tolerance, `visudo -c` error parsing (single, multiple, empty, no-error)
- [x] Resolver internal tests: 3 cases for `pluginsOfType`, including the nil-vs-empty-array distinction
- [x] `go build ./...` clean on `providers/os`
- [x] `go vet` clean on new files
- [x] Codegen (`./lr go`) and versions stamp (`./lr versions`) committed alongside source
- [ ] Manual smoke test against a real Linux host (run `sudo.version` via cnquery shell)
- [ ] Manual smoke test against an AIX host (validate the replacement path for the CIS AIX 7.x query)

## Follow-ups (separate PRs)

- Update `cnspec-enterprise-policies` `queries/aix.mql.yaml:4904` to use `sudo.version` once this lands
- Backfill the same change for any other CIS policy that grew a `sudo --version` shellout

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by tas50** [comment]
Addressed both findings in 95fcfdb:

**Command injection (mondoo-code-review)**
- Wrapped the `sudo -V` path in `shared.ShellEscape` so init-supplied paths can't break out of the command string. Verified against 7 attack payloads (`;`, `&&`, `|`, backticks, `$()`, redirects, spaces).
- Added a filesystem-existence check before invocation so `init(path: "/tmp/anything")` can't run arbitrary binaries even prior to escaping.

**license-check CI**
- Updated copyright headers to the repo's `2024, 2026` format.

Three new unit tests cover the safe-path passthrough, injection-payload escaping, and the empty-path edge case.

**Comment by tas50** [comment]
Pushed e65501088 — addresses 3 correctness findings from a deep review pass (manual + independent code-reviewer agent):

**1. \`installed()\` returned true for non-existent init paths**
`sudo(path: "/tmp/doesnotexist").installed` was returning `true` because `Path` pre-populated via `init` short-circuits the `path()` resolver, and `installed()` only checked `p.Data != ""`. Now verifies filesystem existence via afero.

**2. \`ParseVisudoErrors\` silently dropped the colon-format error variant**
visudo emits both `>>> /etc/sudoers: syntax error near line 42 <<<` and `/etc/sudoers: Invalid Defaults entry: badparam`. The latter was being silently dropped, producing `valid=false, errors=[]`. Now both formats are recognized, `<file>: parsed OK` success markers are filtered explicitly so they don't get mis-matched as colon-format errors, and the resolver synthesizes a fallback error from the raw output when visudo exits non-zero but the parser finds nothing recognized.

**3. Dropped unused \`Group\` family from \`rePluginLine\`**
The alternation never matched real sudo output and would have produced phantom plugin entries from crafted fixtures.

**Tests added: 10**
- 4 resolver-level tests for `installed()` — unset Path, real binary, init-supplied non-existent path (regression), init-supplied path with shell metacharacters (defense in depth).
- 5 parser tests for visudo — colon format, mixed formats, `parsed OK` suppression, partial-success runs, unrecognized-noise dropping.
- 1 regression test that `Group <type> plugin` lines are not recognized.

Resolver-level tests use an in-memory afero fs via a minimal `sudoTestConn` — no changes to the shared LinuxMock fixture.

File: providers/os/resources/sudo.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+// id methods for the private sub-resources.
+
+func (p *mqlSudoPlugin) id() (string, error) {
+	return "sudo.plugin/" + p.Name.Data + ":" + p.Version.Data, nil
+}
+
+func (v *mqlSudoValidation) id() (string, error) {
+	if v.Valid.Data {
+		return "sudo.validation/valid", nil
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/os/resources/sudo.go:377]: Intentional — the resource is a per-asset summary, so within a single scan we want the cached result. Re-running `validate` mid-scan with a different sudoers state isn't a flow the scan engine exercises today; if it ever needs to, we can add a fresh handle then.

**Comment by tas50** [comment]
Closing in favor of #7985, which carries the same contents.
